test/conftest.py

```diff
@@ -237,6 +231,12 @@ def stop_xephyr(self):
 
         self.proc = None
 
+        # clean up the lock file for the display we allocated
```

Comment:
why?

Reply:
Haha! More like why do we have to do this while we didn't with xvfbwrapper?

Reply:
We didn't have to because we were just leaking things everywhere. If you look at my original patch in the other PR, I added cleanup code in the same place.

xcffib.testing.XvfbTest() will handle this for you, but since we're going out of band to use xcffib.testing.find_display() (for use with xephyr, not xvfb), we have to do manual cleanup. The xcffib test suite uses xvfb exclusively, so it doesn't offer a wrapper for xephyr.

It's a little bit unnecessary to use both xephyr and xvfb, we could probably just switch everything to just xephyr. But xvfb is probably faster for the test cases where it works. Xephyr can handle mouse/keyboard inputs, where xvfb really just handles drawing. So for qtile, we really do need xephyr to fake input, whereas the xcffib tests mostly just try to manipulate windows to make sure the protocol basics work correctly.

Reply:
Thanks!